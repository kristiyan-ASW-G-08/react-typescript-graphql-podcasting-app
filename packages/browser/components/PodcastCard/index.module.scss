@import '../../styles/variables.module.scss';
@import '../../styles/mediaQueries.module.scss';

.card {
  display: grid;
  justify-items: center;
  grid-gap: 1rem;
  padding: 2rem;
  border-radius: 5px;
  border: 1px solid $border;
  img {
    height: 10rem;
    width: 10rem;
  }
  h1 {
    font-size: 1.5rem;
  }
  h2 {
    font-size: 1rem;
    color: $border;
  }
}
.buttonContainer {
  display: grid;
  grid-gap: 0.5rem;
  button {
    background: $primary;
    border: none;
    border-radius: 5px;
    font-size: 0.9rem;
    color: $white;
    height: 2rem;
    width: 10rem;
    &:hover {
      background: $white;
      color: $primary;
      border: solid 1px $primary;
    }
  }
}
