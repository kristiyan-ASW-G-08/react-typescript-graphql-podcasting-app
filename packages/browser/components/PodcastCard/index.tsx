import Link from 'next/link';
import { FC } from 'react';
import styles from './index.module.scss';

interface PodcastProps {
  title: string;
  website: string;
  cover: string;
  _id: string;
  userId: string;
  user: {
    username: string;
    _id: string;
  };
}

const PodcastCard: FC<PodcastProps> = ({
  title,
  user,
  website,
  _id,
  cover,
  userId,
}) => {
  return (
    <div className={styles.card}>
      <img src={`${process.env.NEXT_PUBLIC_SERVER_URL}/${cover}`} />
      <h1>{title}</h1>
      <h2>{user.username}</h2>
      <div className={styles.buttonContainer}>
        {userId === user._id ? (
          <>
            <Link href={`/podcast/edit/${_id}`}>
              <button data-testid="edit">Edit</button>
            </Link>
            <Link href={`/create-episode/${_id}`}>
              <button data-testid="create">Create An Episode</button>
            </Link>
          </>
        ) : (
          ''
        )}
        <Link href={website}>
          <button>Website</button>
        </Link>
      </div>
    </div>
  );
};
export default PodcastCard;
