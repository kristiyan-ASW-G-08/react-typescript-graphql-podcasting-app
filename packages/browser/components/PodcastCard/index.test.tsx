import React, { FC } from 'react';
import { render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PodcastCard from '.';

describe('PodcastCard', () => {
  it('render PodcastCard with non matching user', () => {
    const podcast = {
      title: 'SomeTItle',
      website: 'https://somewebsite.testit',
      cover: 'somecover.svg',
      user: {
        username: 'SomeUser',
        _id: 'someId',
      },
      userId: 'someOtherId',
      _id: '1',
    };
    expect.assertions(3);

    const { container, getByTestId, queryByText } = render(
      <PodcastCard {...podcast} />,
    );
    expect(container).toMatchSnapshot();

    const createButton = queryByText('Create An Episode');
    const editButton = queryByText('Edit');

    expect(createButton).not.toBeInTheDocument();
    expect(editButton).not.toBeInTheDocument();
  });

  it('render PodcastCard with matching user', () => {
    const podcast = {
      title: 'SomeTItle',
      website: 'https://somewebsite.testit',
      cover: 'somecover.svg',
      user: {
        username: 'SomeUser',
        _id: 'someId',
      },
      userId: 'someId',
      _id: '1',
    };
    expect.assertions(3);

    const { container, getByTestId } = render(<PodcastCard {...podcast} />);

    expect(container).toMatchSnapshot();

    const createButton = getByTestId('create');
    const editButton = getByTestId('edit');

    expect(createButton).toBeInTheDocument();
    expect(editButton).toBeInTheDocument();
  });
});
