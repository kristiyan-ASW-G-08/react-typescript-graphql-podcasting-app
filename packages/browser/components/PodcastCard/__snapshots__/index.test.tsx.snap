// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcasCArd render PodcastCard with non matching user 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="undefined/somecover.svg"
    />
    <h1>
      SomeTItle
    </h1>
    <h2>
      SomeUser
    </h2>
    <div
      class="buttonContainer"
    >
      <button>
        Website
      </button>
    </div>
  </div>
</div>
`;

exports[`PodcasCArd render PodcastCard with non matching user 2`] = `
<div>
  <div
    class="card"
  >
    <img
      src="undefined/somecover.svg"
    />
    <h1>
      SomeTItle
    </h1>
    <h2>
      SomeUser
    </h2>
    <div
      class="buttonContainer"
    >
      <button
        data-testid="edit"
      >
        Edit
      </button>
      <button
        data-testid="create"
      >
        Create An Episode
      </button>
      <button>
        Website
      </button>
    </div>
  </div>
</div>
`;

exports[`Podcast render PodcastCard with non matching user 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="undefined/somecover.svg"
    />
    <h1>
      SomeTItle
    </h1>
    <h2>
      SomeUser
    </h2>
    <div
      class="buttonContainer"
    >
      <button>
        Website
      </button>
    </div>
  </div>
</div>
`;

exports[`Podcast render PodcastCard with non matching user 2`] = `
<div>
  <div
    class="card"
  >
    <img
      src="undefined/somecover.svg"
    />
    <h1>
      SomeTItle
    </h1>
    <h2>
      SomeUser
    </h2>
    <div
      class="buttonContainer"
    >
      <button
        data-testid="edit"
      >
        Edit
      </button>
      <button
        data-testid="create"
      >
        Create An Episode
      </button>
      <button>
        Website
      </button>
    </div>
  </div>
</div>
`;

exports[`PodcastCard render PodcastCard with matching fuser 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="undefined/somecover.svg"
    />
    <h1>
      SomeTItle
    </h1>
    <h2>
      SomeUser
    </h2>
    <div
      class="buttonContainer"
    >
      <button
        data-testid="edit"
      >
        Edit
      </button>
      <button
        data-testid="create"
      >
        Create An Episode
      </button>
      <button>
        Website
      </button>
    </div>
  </div>
</div>
`;

exports[`PodcastCard render PodcastCard with matching user 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="undefined/somecover.svg"
    />
    <h1>
      SomeTItle
    </h1>
    <h2>
      SomeUser
    </h2>
    <div
      class="buttonContainer"
    >
      <button
        data-testid="edit"
      >
        Edit
      </button>
      <button
        data-testid="create"
      >
        Create An Episode
      </button>
      <button>
        Website
      </button>
    </div>
  </div>
</div>
`;

exports[`PodcastCard render PodcastCard with non matching user 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="undefined/somecover.svg"
    />
    <h1>
      SomeTItle
    </h1>
    <h2>
      SomeUser
    </h2>
    <div
      class="buttonContainer"
    >
      <button>
        Website
      </button>
    </div>
  </div>
</div>
`;

exports[`PodcastCard render PodcastCard with non matching user 2`] = `
<div>
  <div
    class="card"
  >
    <img
      src="undefined/somecover.svg"
    />
    <h1>
      SomeTItle
    </h1>
    <h2>
      SomeUser
    </h2>
    <div
      class="buttonContainer"
    >
      <button
        data-testid="edit"
      >
        Edit
      </button>
      <button
        data-testid="create"
      >
        Create An Episode
      </button>
      <button>
        Website
      </button>
    </div>
  </div>
</div>
`;

exports[`PodcastCard render PodcastCard with user 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="undefined/somecover.svg"
    />
    <h1>
      SomeTItle
    </h1>
    <h2>
      SomeUser
    </h2>
    <div
      class="buttonContainer"
    >
      <button
        data-testid="edit"
      >
        Edit
      </button>
      <button
        data-testid="create"
      >
        Create An Episode
      </button>
      <button>
        Website
      </button>
    </div>
  </div>
</div>
`;
